src/python/pants/backend/codegen/write_codegen_goal.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+
+from pants.core.util_rules.distdir import DistDir
+from pants.engine.fs import Digest, MergeDigests, Workspace
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.rules import Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import (
+    GenerateSourcesRequest,
+    HydratedSources,
+    HydrateSourcesRequest,
+    RegisteredTargetTypes,
+    Sources,
+    Targets,
+)
+from pants.engine.unions import UnionMembership
+
+logger = logging.getLogger(__name__)
+
+
+class WriteCodegenSubsystem(GoalSubsystem):
+    """Write generated files to `dist/codegen` for use outside of Pants."""
+
+    name = "write-codegen"
```

Comment:
Please feel free to bikeshed the goal name. In v1, this was `gen`. I wanted a longer name to emphasize that this is not necessary to run for codegen to work; this is only for external consumption.

Reply:
Maybe `export-codegen`?

Reply:
+1 for `export-codegen`